### PR TITLE
add-security-merge-to-master

### PR DESCRIPTION
**Pull request by alwayswannajava**

### CODE REVIEW

File: src/main/java/com/spring/onlinebookstore/security/CustomUserDetailsService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException(username));
```
Comment (Elena-Bruyako) [inline src/main/java/com/spring/onlinebookstore/security/CustomUserDetailsService.java]: provide informative message

```suggestion
                .orElseThrow(() -> new EntityNotFoundException(username));
```

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 2]
```diff
@@ -38,4 +38,10 @@ databaseChangeLog:
                   name: shipping_address
                   type: varchar(255)
                   constraints:
-                    nullable: true
\ No newline at end of file
+                    nullable: true
+              - column:
+                  name: role_id
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]: remove, user doesn't have role id column

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**